tests/ludwig/encoders/test_generic_encoders.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+import torch
+
+from ludwig.encoders.generic_encoders import DenseEncoder, PassthroughEncoder
+
+
+@pytest.mark.parametrize("input_size", [1, 2, 10])
+@pytest.mark.parametrize("categorical", [True, False])
+def test_generic_passthrough_encoder(input_size: int, categorical: bool):
+    passthrough_encoder = PassthroughEncoder(input_size)
+    # Passthrough encoder allows categorical input feature (int), dense encoder's input must be float.
+    if categorical:
+        inputs = torch.randint(10, (10, input_size))
+    else:
+        inputs = torch.rand((10, input_size))
+    outputs = passthrough_encoder(inputs)
+    assert "encoder_output" in outputs
+    encoder_output = outputs["encoder_output"]
+    # Ensures output shape matches encoder expected output shape.
+    assert encoder_output.shape[1:] == passthrough_encoder.output_shape
+
+
+@pytest.mark.parametrize("input_size", [1, 2, 10])
+@pytest.mark.parametrize("num_layers", [1, 3, 6])
+@pytest.mark.parametrize("fc_size", [1, 2, 10, 256])
+def test_generic_dense_encoder(input_size: int, num_layers: int, fc_size: int):
+    dense_encoder = DenseEncoder(input_size, num_layers=num_layers, fc_size=fc_size)
+    inputs = torch.rand((10, input_size))
+    outputs = dense_encoder(inputs)
+    assert "encoder_output" in outputs
```

Comment:
Does it make sense to simplify this to:

```
assert outputs["encoder_output"].shape[1:] == dense_encoder.output_shape
```

Failures due to a missing "encoder_output" on this simplified form are still quite descriptive.

Reply:
I prefer to err on the side of explicitness and 'one claim per assertion', but thats just my style preference I could go either way.

Reply:
The concision of having a single assert for one behavior seems nice, but I agree that it's also less explicit. Happy to go with what you and other people think is best here.